analyzers/src/SonarAnalyzer.CFG/Helpers/StringExtensions.cs

```diff
@@ -31,25 +31,25 @@ public static class StringExtensions
         /// Sequence of upper case letters is considered as single word.
         ///
         /// For example:
-        /// thisIsAName => this is a name
-        /// ThisIsSMTPName => this is smtp name
-        /// bin2hex => bin hex
-        /// HTML => html
-        /// SOME_value => some value
-        /// PEHeader => pe header
+        /// <list type="table">
+        /// <item>thisIsAName => ["THIS", "IS", "A", "NAME"]</item>
+        /// <item>ThisIsSMTPName => ["THIS", "IS", "SMTP", "NAME"]</item>
+        /// <item>bin2hex => ["BIN", "HEX"]</item>
+        /// <item>HTML => ["HTML"]</item>
+        /// <item>SOME_value => ["SOME", "VALUE"]</item>
+        /// <item>PEHeader => ["PE", "HEADER"]</item>
+        /// </list>
         /// </summary>
         /// <param name="name">A string containing words.</param>
-        /// <returns>A list of words (all lowercase) contained in the string.</returns>
+        /// <returns>A list of words (all uppercase) contained in the string.</returns>
         public static IEnumerable<string> SplitCamelCaseToWords(this string name)
         {
-            bool IsFollowedByLower(int i) => i + 1 < name.Length && char.IsLower(name[i + 1]);
-
             if (name == null)
             {
                 yield break;
             }
 
-            var currentWord = new StringBuilder();
+            var currentWord = new StringBuilder(capacity: name.Length);
```

Comment:
This change should not be needed. The names will never be long enough to make a significant difference.